DNN Platform/Library/Services/Authentication/AuthenticationController.cs

```diff
@@ -268,6 +274,32 @@ public static List<AuthenticationInfo> GetEnabledAuthenticationServices()
             }
             return enabled;
         }
+        
+        public static bool HasSocialAuthenticationEnabled(UserControl control = null)
+        {
+            return (from a in GetEnabledAuthenticationServices()
+                    let enabled = (a.AuthenticationType.Equals("Facebook")
+                                     || a.AuthenticationType.Equals("Google")
+                                     || a.AuthenticationType.Equals("Live")
+                                     || a.AuthenticationType.Equals("Twitter"))
+                                  ? IsEnabledForPortal(a, PortalSettings.Current.PortalId)
+                                  : !string.IsNullOrEmpty(a.LoginControlSrc) && ((control?.LoadControl("~/" + a.LoginControlSrc) as AuthenticationLoginBase)?.Enabled ?? true)
+                    where !a.AuthenticationType.Equals("DNN") && enabled
+                    select a).Any();
+        }
+
+        /// <summary>
+        /// Determines whether [is enabled for portal] [the specified authentication].

```

Comment:
@JoshuaBradley can you fix the doc comments (and add some to the above method)?  Thanks!